### PR TITLE
New ping

### PR DESCRIPTION
**Pull request by callan-stretton**

### CODE REVIEW

File: client/components/NewPing.jsx [hunk 1]
```diff
@@ -5,14 +5,57 @@ class NewPing extends React.Component {
   constructor (props) {
     super(props)
     this.state = {
-
+      recepientId: null,
+      image: null,
```
Comment (harrison-symes) [inline client/components/NewPing.jsx]: rename to image_url for db

File: client/components/NewPing.jsx [hunk 2]
```diff
@@ -5,14 +5,57 @@ class NewPing extends React.Component {
...
+    this.selectImage = this.selectImage.bind(this)
+    this.submitInfo = this.submitInfo.bind(this)
+  }
+
+  selectRecepient (evt) {
+    this.setState({recepientId: Number(evt.target.value)})
   }
+
+  selectImage (evt) {
+    this.setState({image: evt.target.value})
```
Comment (harrison-symes) [inline client/components/NewPing.jsx]: image_url

File: client/components/NewPing.jsx [hunk 3]
```diff
@@ -5,14 +5,57 @@ class NewPing extends React.Component {
...
+    console.log(this.state) // shows what's being posted
+  }
+
   render () {
+    const {users, currentUser} = this.props
+    const recepient = users.find(user => user.id == this.state.recepientId)
     return (
-      <div><h1>New Ping will go here</h1></div>
+      <div>
+        <img src={currentUser.photo_url} /> {/*this is the sender image*/}
```
Comment (harrison-symes) [inline client/components/NewPing.jsx]: Remove comments

File: client/components/NewPing.jsx [hunk 4]
```diff
@@ -5,14 +5,57 @@ class NewPing extends React.Component {
...
+        <img src={this.state.image || 'http://cumbrianrun.co.uk/wp-content/uploads/2014/02/default-placeholder-300x300.png'} /> {/*this is the image being pinged*/}
+        <img src={recepient ? recepient.photo_url : 'https://www.wsxenterprise.co.uk/wp-content/uploads/2015/11/blank-profile-picture-973461_1280-300x300.png'} /> {/*this is the recepient image*/}
```
Comment (harrison-symes) [inline client/components/NewPing.jsx]: Remove comments

File: client/components/NewPing.jsx [hunk 5]
```diff
@@ -5,14 +5,57 @@ class NewPing extends React.Component {
...
+        <form onSubmit={this.submitInfo}>
+          Image: <input type="text" onChange={this.selectImage}/><br/>
+          <select onChange={this.selectRecepient} name="" className="selector">
+            <option selected disabled >Select user</option>
+            {users.map(user => {
+              return (<option key={user.id} value={user.id}>{user.name}</option>)
+            })}
+          </select>
+          <br/>
```
Comment (harrison-symes) [inline client/components/NewPing.jsx]: remove <br />

File: client/components/NewPing.jsx [hunk 6]
```diff
@@ -5,14 +5,57 @@ class NewPing extends React.Component {
...
+    const {users, currentUser} = this.props
+    const recepient = users.find(user => user.id == this.state.recepientId)
     return (
-      <div><h1>New Ping will go here</h1></div>
+      <div>
+        <img src={currentUser.photo_url} /> {/*this is the sender image*/}
+        <img src={this.state.image || 'http://cumbrianrun.co.uk/wp-content/uploads/2014/02/default-placeholder-300x300.png'} /> {/*this is the image being pinged*/}
+        <img src={recepient ? recepient.photo_url : 'https://www.wsxenterprise.co.uk/wp-content/uploads/2015/11/blank-profile-picture-973461_1280-300x300.png'} /> {/*this is the recepient image*/}
+        <form onSubmit={this.submitInfo}>
+          Image: <input type="text" onChange={this.selectImage}/><br/>
```
Comment (harrison-symes) [inline client/components/NewPing.jsx]: Use <label> tags

File: client/components/Register.jsx [hunk 7]
```diff
@@ -40,9 +40,9 @@ class Register extends React.Component {
           </label>
           <label>photo
             <input type='text' onChange={this.handleChange} placeholder="url..." />
-          </label>
+          </label>?
```
Comment (harrison-symes) [inline client/components/Register.jsx]: Is this supposed to be here?

File: client/components/SignIn.jsx [hunk 8]
```diff
@@ -15,26 +16,26 @@ class SignIn extends React.Component {
       [e.target.name]: e.target.value
     })
   }
+  render () {
+    return (
+      <div className='signin'>
+        <div className='signin-form'>
```
Comment (harrison-symes) [inline client/components/SignIn.jsx]: unneeded div

**Review by harrison-symes — CHANGES_REQUESTED** [review]

File: client/components/NewPing.jsx [hunk 9]
```diff
@@ -5,14 +5,58 @@ class NewPing extends React.Component {
   constructor (props) {
     super(props)
     this.state = {
-
+      recepientId: null,
+      image_url: null,
```
Comment (don-smith) [inline client/components/NewPing.jsx]: `image_url` ➡️ `imageUrl`

File: client/components/NewPing.jsx [hunk 10]
```diff
@@ -5,14 +5,58 @@ class NewPing extends React.Component {
...
+    this.selectImage = this.selectImage.bind(this)
+    this.submitInfo = this.submitInfo.bind(this)
+  }
+
+  selectRecepient (evt) {
+    this.setState({recepientId: Number(evt.target.value)})
   }
+
+  selectImage (evt) {
+    this.setState({image_url: evt.target.value})
```
Comment (don-smith) [inline client/components/NewPing.jsx]: image_url ➡️ imageUrl

File: client/components/NewPing.jsx [hunk 11]
```diff
@@ -5,14 +5,58 @@ class NewPing extends React.Component {
...
-      <div><h1>New Ping will go here</h1></div>
+      <div>
+        <img src={currentUser.photo_url} />
+        <img src={this.state.image_url || 'http://cumbrianrun.co.uk/wp-content/uploads/2014/02/default-placeholder-300x300.png'} />
+        <img src={recepient ? recepient.photo_url : 'https://www.wsxenterprise.co.uk/wp-content/uploads/2015/11/blank-profile-picture-973461_1280-300x300.png'} />
+        <form onSubmit={this.submitInfo}>
+          <label>Image:
+            <input type="text" onChange={this.selectImage}/>
+          </label>
+          <select onChange={this.selectRecepient} name="" className="selector">
```
Comment (don-smith) [inline client/components/NewPing.jsx]: Remove `name` attribute if we aren't going to use it.

**Review by don-smith — CHANGES_REQUESTED** [review]

**Pull request merged**